doc: update release notes and upgrade docs for v1.14 (#13997)
rook/rook: Documentation/CRDs/Cluster/network-providers.md

    need to be an order of magnitude larger (or more) than the host address space to allow the
    storage cluster to grow in the future.

> If these prerequisites are not achievable, plan to set the Rook operator config

Reviewer: even more direct...
```suggestion
If these prerequisites are not achievable, set the Rook operator config
```

Author: We don't actually want the users to set the value immediately, so I used "plan to set" instead of just "set". The setting is handled later.

    }'
```

## Holder Pod Deprecation

> Rook plans to deprecate CSI "holder" pods in Rook v1.16. Clusters with `csi-*plugin-holder-*` pods

Reviewer: We are planning to remove support for them completely, right? I think "deprecate" means different things to different people, so just to clarify.
```suggestion
Rook plans to remove CSI "holder" pods in Rook v1.16. Clusters with `csi-*plugin-holder-*` pods
```

Author: Okay. I changed this throughout all docs

present in the Rook operator namespace must plan to set `CSI_DISABLE_HOLDER_PODS` to `"true"` after
Rook v1.14 is installed and before v1.16 is installed by following the migration sections below.

> CephClusters that do not use `network.provider: multus` can follow the

Reviewer: My confusion before was that all clusters without multus would need to go to that section. But the point is that only clusters with holder pods need to do it, right?
```suggestion
Clusters with holder pods enabled that do not use `network.provider: multus` can follow the
```

Author: It's mentioned immediately above that clusters with holder pods must migrate.

> Clusters with `csi-*plugin-holder-*` pods
present in the Rook operator namespace must plan to set `CSI_DISABLE_HOLDER_PODS` to `"true"` after
Rook v1.14 is installed and before v1.16 is installed by following the migration sections below.

> CephClusters that do not use `network.provider: multus` can skip to the

Reviewer: They can skip the section completely, right?
```suggestion
CephClusters that do not use `network.provider: multus` can skip the
```

Author: "skip to" is what I intended, but I can see this might be confused easily, so I changed it to "follow"

[Disabling Holder Pods](#disabling-holder-pods) section.

CephClusters that use `network.provider: multus` will need to plan the migration more carefully.
Read the [Disabling Holder Pods with Multus and CSI Host Networking](#disabling-holder-pods-with-multus-and-csi-host-networking)

> section. Decide whether to use CSI host networking or not following the sections below.

Reviewer: > Decide whether to use CSI host networking or not following the sections below.

Without holder pods, won't we always require CSI host networking? Or maybe I'm misinterpreting this sentence.

Author: `CSI_ENABLE_HOST_NETWORKING` is not required to be true. But we did talk about making another update after this PR to make that situation unsupported.

Reviewer: Sounds good to follow up in a separate PR

steps above before continuing.

**Step 2**
Begin by setting `CSI_DISABLE_HOLDER_PODS: "true"` -- also set the desired value of

> `CSI_ENABLE_HOST_NETWORK` if needed.

Reviewer: Again, isn't `CSI_ENABLE_HOST_NETWORK: true` required when holder pods are disabled? Or how do they know if it's needed?

Author: They can read the networking section on CSI host networking to decide if they want to enable it or not. It's not required. But we did talk about making another update after this PR to make that situation unsupported.
